docs/blueprints/EP031.rst

```diff
@@ -0,0 +1,273 @@
+:EP: 31
+:Title: Enabling Telemetry for EVCs provisioned with Kytos-ng mef-eline
+:Authors:
+    - Jeronimo Bezerra <jbezerra AT fiu DOT edu>
+    - Italo Valcy <idasilva AT fiu DOT edu>
+    - Vinicius Arcanjo <vindasil AT fiu DOT edu>
+:Created: 2022-08-24
+:Kytos-Version: 2022.3
+:Status: Draft
+
+****************************************
+EP031 - Telemetry-enabled EVCs
+****************************************
+
+
+Abstract
+========
+
+Blueprint 31 presents requirements and features to support In-band Network Telemetry (a.k.a. INT or telemetry) for EVCs created by the Kytos **mef-eline** napp.
+
+
+Motivation
+==========
+
+INT enables per-packet monitoring and it is specified by the P4.org consortium as one of P4 use cases. To enable full network visibility, all EVCs must support INT. For the network devices available at AmLight, INT will be performed by the network data plane when instructed to do so via OpenFlow NoviFlow Experimenter's actions.
+
+Supporting INT with the current implementation of the NoviFlow's NoviWare OS won't be trivial due to many conditions imposed by NoviWare. This blueprint will focus on enabling a MINIMALLY functional support for INT over Kytos. Once this blueprint is implemented, new blueprints will be created addressing more complex requirements.
+
+
+I. Requirements
+===============
+
+This blueprint has the following characteristics:
+
+  1. There will be no concerns about number or location of the INT Hops switches. Currently, an INT Sink switch can only export up to 10 metadata stacked by the network.
+  2. There will be no concerns about MTU restrictions. NoviWare and Tofino support frames as large as 10KB which is more than enough. However, any legacy device in the middle must be previously identified since legacy devices usually have a MTU of up to 9216 Bytes.
+  3. There will be no concerns if INT Sink switches have loops for INT reports (Section III). This blueprint assumes physical loops are already deployed by the operators during the commissioning phase.
+  4. There will be no concerns about proxy ports and their mappings.
```

Comment:
How this Napp will keep track of the proxy ports? Which one is being used by each EVC/UNI/etc?

Reply:
It won't. Proxy ports will be identified (by looking for intra-switch loops) and the first loop will be always used until we decide what to do from the management perspective. This is subject to another blueprint since it has several implications (number of proxies, ownership, number of proxies supported per box, etc.)

Reply:
I believe this comment was addressed by the Proxy Port blueprint, right @jab1982 ? I will mark it as resolved, but please feel free to reopen the discussion.